internal/pod_models.go

```diff
@@ -982,3 +982,93 @@ func GetObjectMetadata(cluster *fdbv1beta2.FoundationDBCluster, base *metav1.Obj
 func GetPodDNSName(cluster *fdbv1beta2.FoundationDBCluster, podName string) string {
 	return fmt.Sprintf("%s.%s.%s.svc.%s", podName, cluster.Name, cluster.Namespace, cluster.GetDNSDomain())
 }
+
+// GetProfileAnalyzer builds a deployment for profile analyzing.
+func GetProfileAnalyzer(namespace string, cmd []string, args []string) (*appsv1.Deployment, error) {
+	deploymentName := "fdb-profile-analyzer"
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   namespace,
+			Name:        deploymentName,
+			Annotations: map[string]string{},
+			Labels:      map[string]string{},
+		},
+	}
+
+	podTemplate := &corev1.PodTemplateSpec{}
+
+	var mainContainer *corev1.Container
+	var initContainer *corev1.Container
+
+	if mainContainer == nil {
```

Comment:
Why should the mainContainer not. be nil if we just created it a few lines above? I think most of the creation of the struct can be merged into one step.